Tests/MuxUploadSDKTests/Upload Tests/ReporterTests.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  ReporterTests.swift
+//
+
+import Foundation
+import XCTest
+
+@testable import MuxUploadSDK
+
+class ReporterTests: XCTestCase {
+
+    enum ExpectedJSONStrings {
+        static let inputStandardizationFailed = #"{"data":{"app_name":"AcmeApp","app_version":"3.2.1","device_model":"iPad","error_description":"foo","input_duration":3.1400000000000001,"input_size":1500000,"maximum_resolution":"default","non_standard_input_reasons":[],"platform_name":"iPadOS","platform_version":"15.0.0","sdk_version":"0.4.1","standardization_end_time":"2023-07-07T03:43:58Z","standardization_start_time":"2023-07-07T03:38:58Z","upload_canceled":false,"upload_url":"https:\/\/www.example.com"},"session_id":"xyz789","type":"upload_input_standardization_failed","version":"1"}"#
+
+        static let inputStandardizationSucceeded = #"{"data":{"app_name":"AcmeApp","app_version":"3.2.1","device_model":"iPad","input_duration":3.1400000000000001,"input_size":1500000,"maximum_resolution":"default","non_standard_input_reasons":[],"platform_name":"iPadOS","platform_version":"15.0.0","sdk_version":"0.4.1","standardization_end_time":"2023-07-07T03:43:58Z","standardization_start_time":"2023-07-07T03:38:58Z","upload_url":"https:\/\/www.example.com"},"session_id":"jkl567","type":"upload_input_standardization_succeeded","version":"1"}"#
+
+        static let uploadFailed = #"{"data":{"app_name":"AcmeApp","app_version":"3.2.1","device_model":"iPad","error_description":"foo","input_duration":3.1400000000000001,"input_size":1500000,"input_standardization_requested":false,"platform_name":"iPadOS","platform_version":"16.2.0","region_code":"US","sdk_version":"0.3.0","upload_end_time":"2023-07-07T04:12:48Z","upload_start_time":"2023-07-07T04:12:18Z","upload_url":"https:\/\/www.example.com"},"session_id":"abc123","type":"upload_failed","version":"1"}"#
+
+        static let uploadSucceeded = #"{"data":{"app_name":"AcmeApp","app_version":"3.2.1","device_model":"iPad","input_duration":3.1400000000000001,"input_size":1500000,"input_standardization_requested":true,"platform_name":"iPadOS","platform_version":"16.2.0","region_code":"US","sdk_version":"0.3.0","upload_end_time":"2023-07-07T04:12:48Z","upload_start_time":"2023-07-07T04:12:18Z","upload_url":"https:\/\/www.example.com"},"session_id":"abc123","type":"upload_succeeded","version":"1"}"#
+    }
+
+    var jsonEncoder = Reporter().jsonEncoder
+
+    func testInputStandardizationFailedEventSerialization() throws {
+        let data = InputStandardizationFailedEvent.Data(
+            appName: "AcmeApp",
+            appVersion: "3.2.1",
+            deviceModel: "iPad",
+            errorDescription: "foo",
+            inputDuration: 3.14,
+            inputSize: 1_500_000,
+            maximumResolution: "default",
+            nonStandardInputReasons: [],
+            platformName: "iPadOS",
+            platformVersion: "15.0.0",
+            sdkVersion: "0.4.1",
```

Comment:
Right?
```suggestion
            sdkVersion: "0.4.0",
```

Reply:
This is test code, the version used here is immaterial